### ISSUE TITLE
ArrayIndexOutOfBoundsException in determineCurrentVersion for git

### ISSUE THREAD

**Issue by tulinkry** [oracle/opengrok#2404]
I recently received an exception, which eventually hides one of my projects (because the repository information is missing)

```
09-Oct-2018 12:17:59.478 WARNING [invalidate-repos-18] org.opengrok.indexer.history.HistoryGuru$3.run Could not create git for '/var/opengrok/src/project
 java.lang.ArrayIndexOutOfBoundsException: 32
	at sun.util.calendar.BaseCalendar.getCalendarDateFromFixedDate(BaseCalendar.java:453)
	at java.util.GregorianCalendar.computeFields(GregorianCalendar.java:2397)
	at java.util.GregorianCalendar.computeFields(GregorianCalendar.java:2312)
	at java.util.Calendar.setTimeInMillis(Calendar.java:1804)
	at java.util.Calendar.setTime(Calendar.java:1770)
	at java.text.SimpleDateFormat.format(SimpleDateFormat.java:943)
	at java.text.SimpleDateFormat.format(SimpleDateFormat.java:936)
	at java.text.DateFormat.format(DateFormat.java:345)
	at org.opengrok.indexer.history.GitRepository.determineCurrentVersion(GitRepository.java:725)
	at org.opengrok.indexer.history.RepositoryFactory.getRepository(RepositoryFactory.java:150)
	at org.opengrok.indexer.history.RepositoryFactory.getRepository(RepositoryFactory.java:187)
	at org.opengrok.indexer.history.HistoryGuru$3.run(HistoryGuru.java:948)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Last 10 git log records look like this:
```log
2018-08-31T15:47:34-04:00#202af7b Name Surname BUGFIX: Lorem Ipsum
2018-08-27T10:32:42-04:00#317495a Name Surname FEATURE: Lorem Ipsum
2018-08-20T09:54:42-04:00#ddf6817 Name Surname Merge "FEATURE: Lorem Ipsum"
2018-06-12T09:12:54-04:00#15c9c63 Name Surname FEATURE: Lorem Ipsum
2018-06-06T11:03:55-04:00#1d4f632 Name Surname FEATURE: Lorem Ipsum
2018-01-19T11:27:35-05:00#0003062 Name Surname REVERT: Lorem Ipsum
2018-01-18T14:42:14-05:00#70a2ac1 Name Surname Merge "REFACTOR: Lorem Ipsum"
2018-01-18T14:32:09-05:00#079d4fc Name Surname REFACTOR: Lorem Ipsum
2018-01-12T12:14:22-05:00#c7dae34 Name Surname FEATURE: Lorem Ipsum
2018-01-09T11:20:44-05:00#7c61a82 Name Surname REFACTOR: Lorem Ipsum
```

The text of the log message doesn't seem too important for me in this.

```bash
-bash-4.1$ /opt/git/git283/bin/git --version
git version 2.8.3
```

The problem here looks like the outputDateFromat. That is the exception place.

**Comment by tulinkry** [comment]
Also when the repository info is missing (after this exception), not even a reindex can get it back. The refresh repositories always refreshes existing repositories (and does not discover new ones). So once this happens the project is gone for good.

**Comment by tulinkry** [comment]
I don't think so. It just happened once and doesn't happen again so I guess this is hard to reproduce.

**Comment by tulinkry** [comment]
But we can do something at least with the repository discovery during the refresh..?

**Comment by vladak** [comment]
Yes, that would be separate issue however.

Dne čt 11. 10. 2018 17:37 uživatel Kryštof Tulinger <
notifications@github.com> napsal:

**Comment by tulinkry** [comment]
Fixed in #2570
